cmd/livekit-cli/sip.go

```diff
@@ -283,62 +423,74 @@ func listSipDispatchRule(c *cli.Context) error {
+	})
+}
 
-	if c.Bool("verbose") {
-		PrintJSON(res)
+func deleteSIPDispatchRule(c *cli.Context) error {
+	cli, err := createSIPClient(c)
+	if err != nil {
+		return err
 	}
-
-	return nil
+	return forEachID(c, func(ctx context.Context, id string) error {
+		info, err := cli.DeleteSIPDispatchRule(c.Context, &livekit.DeleteSIPDispatchRuleRequest{
+			SipDispatchRuleId: id,
+		})
+		if err != nil {
+			return err
+		}
+		printSIPDispatchRuleID(info)
+		return nil
+	})
 }
 
-func deleteSIPDispatchRule(c *cli.Context) error {
-	info, err := sipClient.DeleteSIPDispatchRule(context.Background(), &livekit.DeleteSIPDispatchRuleRequest{
+func deleteSIPDispatchRuleLegacy(c *cli.Context) error {
+	cli, err := createSIPClient(c)
+	if err != nil {
+		return err
+	}
+	info, err := cli.DeleteSIPDispatchRule(c.Context, &livekit.DeleteSIPDispatchRuleRequest{
 		SipDispatchRuleId: c.String("id"),
 	})
 	if err != nil {
 		return err
 	}
-
-	printSIPDispatchRuleInfo(info)
+	printSIPDispatchRuleID(info)
 	return nil
 }
 
-func printSIPDispatchRuleInfo(info *livekit.SIPDispatchRuleInfo) {
+func printSIPDispatchRuleID(info *livekit.SIPDispatchRuleInfo) {
 	fmt.Printf("SIPDispatchRuleID: %v\n", info.SipDispatchRuleId)
 }
 
 func createSIPParticipant(c *cli.Context) error {
-	reqFile := c.String("request")
-	reqBytes, err := os.ReadFile(reqFile)
-	if err != nil {
-		return err
-	}
-
-	req := &livekit.CreateSIPParticipantRequest{}
-	err = protojson.Unmarshal(reqBytes, req)
+	cli, err := createSIPClient(c)
 	if err != nil {
 		return err
 	}
+	return createAndPrintReqs(c, func(ctx context.Context, req *livekit.CreateSIPParticipantRequest) (*livekit.SIPParticipantInfo, error) {
+		// CreateSIPParticipant will wait for LiveKit Participant to be created and that can take some time.
+		// Default deadline is too short, thus, we must set a higher deadline for it.
+		ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
+		defer cancel()
+
+		return cli.CreateSIPParticipant(ctx, req)
+	}, printSIPParticipantInfo)
+}
 
-	if c.Bool("verbose") {
-		PrintJSON(req)
-	}
-
-	// CreateSIPParticipant will wait for LiveKit Participant to be created and that can take some time.
-	// Thus, we must set a higher deadline for it.
-	ctx, cancel := context.WithTimeout(c.Context, 30*time.Second)
-	defer cancel()
-
-	info, err := sipClient.CreateSIPParticipant(ctx, req)
+func createSIPParticipantLegacy(c *cli.Context) error {
+	cli, err := createSIPClient(c)
 	if err != nil {
 		return err
 	}
-
-	printSIPParticipantInfo(info)
-	return nil
+	return createAndPrintLegacy(c, func(ctx context.Context, req *livekit.CreateSIPParticipantRequest) (*livekit.SIPParticipantInfo, error) {
+		// CreateSIPParticipant will wait for LiveKit Participant to be created and that can take some time.
+		// Default deadline is too short, thus, we must set a higher deadline for it.
+		ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
```

Comment:
Probably outside my wheelhouse, but when docs are updated it may be good to include constants like this if it is invariant, or expose it as a parameter if not.

Reply:
You are right. I can actually bake it into the SDK, so not configuration will be needed.